libs/dh/wholesale/domain/src/grid-areas.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DateRange, GridAreaDto } from '@energinet-datahub/dh/shared/domain';
+import { isAfter, isBefore, isEqual } from 'date-fns';
+
+export function filterValidGridAreas(
+  gridAreas: GridAreaDto[],
+  dateRange: DateRange | null
+): GridAreaDto[] {
+  if (dateRange === null) return gridAreas;
+  return gridAreas?.filter((gridArea) => {
```

Comment:
nitpick: the function definition says `gridAreas` cannot be empty, so this shouldn't be neccessary:
```suggestion
  return gridAreas.filter((gridArea) => {
```